merge-sort.js

```diff
@@ -0,0 +1,26 @@
+// Marge sort
+
+function mergeSort(array) {
+  if (array.length < 2) {
+    return array;
+  }
+
+  let mid = Math.floor(array.length / 2);
+  let left = mergeSort(array.slice(0, mid));
+  let right = mergeSort(array.slice(mid));
+
+  return merge(left, right);
+}
+
+function merge(left, right) {
+  let result = [];
+
+  while (left.length > 0 && right.length > 0) {
+    result.push(left[0] < right[0] ? left.shift() : right.shift())
```

Comment:
It's better not to change these arrays. Try not to use shift() or other methods that are changing the array. You should just get elements from these arrays with `[]`.